client/blocks/signup-form/passwordless.jsx

```diff
@@ -0,0 +1,258 @@
+import { Button } from '@automattic/components';
+import emailValidator from 'email-validator';
+import { localize } from 'i18n-calypso';
+import PropTypes from 'prop-types';
+import { Component } from 'react';
+import { connect } from 'react-redux';
+import FormLabel from 'calypso/components/forms/form-label';
+import FormTextInput from 'calypso/components/forms/form-text-input';
+import LoggedOutForm from 'calypso/components/logged-out-form';
+import LoggedOutFormFooter from 'calypso/components/logged-out-form/footer';
+import Notice from 'calypso/components/notice';
+import { recordRegistration } from 'calypso/lib/analytics/signup';
+import wpcom from 'calypso/lib/wp';
+import ValidationFieldset from 'calypso/signup/validation-fieldset';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { saveSignupStep, submitSignupStep } from 'calypso/state/signup/progress/actions';
+
+class PasswordlessSignupForm extends Component {
+	static propTypes = {
+		locale: PropTypes.string,
+	};
+
+	static defaultProps = {
+		locale: 'en',
+	};
+
+	state = {
+		isSubmitting: false,
+		email: this.props.step && this.props.step.form ? this.props.step.form.email : '',
+		errorMessages: null,
+	};
+
+	submitTracksEvent = ( isSuccessful, props ) => {
+		const tracksEventName = isSuccessful
+			? 'calypso_signup_actions_onboarding_passwordless_login_success'
+			: 'calypso_signup_actions_onboarding_passwordless_login_error';
+		this.props.recordTracksEvent( tracksEventName, {
+			...props,
+		} );
+	};
+
+	onFormSubmit = async ( event ) => {
+		event.preventDefault();
+
+		if ( ! this.state.email || ! emailValidator.validate( this.state.email ) ) {
+			this.setState( {
+				errorMessages: [ this.props.translate( 'Please provide a valid email address.' ) ],
+				isSubmitting: false,
+			} );
+			this.submitTracksEvent( false, { action_message: 'Please provide a valid email address.' } );
+			return;
+		}
+
+		this.setState( {
+			isSubmitting: true,
+		} );
+
+		// Save form state in a format that is compatible with the standard SignupForm used in the user step.
+		const form = {
+			firstName: '',
+			lastName: '',
+			email: this.state.email,
+			username: '',
+			password: '',
+		};
+
+		this.props.saveSignupStep( {
+			stepName: this.props.stepName,
+			form,
+		} );
+
+		try {
+			const response = await wpcom.undocumented().usersNew(
```

Comment:
Uggh, looks like https://github.com/Automattic/wp-calypso/pull/58109 removed the `usersNew()` undocumented method. So, the signup is no longer working with this method. Could you please change this to the `wpcom.req.post` method like it's used here?

https://github.com/Automattic/wp-calypso/blob/02b621f7bf67a2cdb4353bd1e05e5a9cca0a1bb9/client/lib/signup/step-actions/index.js#L645-L647